### DISCUSSION TITLE
Data List with Paging?

### DISCUSSION THREAD

**Discussion by Antoshjkee** [leekelleher/umbraco-contentment#107] [Idea]
Is there any way I can add paging to the data list? Or you planning to do one for later releases?

**Comment by leekelleher** [comment]
Hi @Antoshjkee. I don't currently have any plans for adding pagination to Data List editors, it's a good suggestion.

It did come up in conversation recently on https://github.com/leekelleher/umbraco-contentment/issues/101#issuecomment-825630548.

I'm wondering, would this only be a feature for the Item Picker editor? as I'm not sure it'd be expected for the others (e.g. checkbox list, radiobutton list, etc).

A current limitation is that the Data List items are fetched as part of the initial Content editing payload, meaning that there aren't any extra Ajax/HTTP requests made to the server to get the items, which would be needed for pagination. _(That's purely something more for me to consider)._ 🤔 

I'll give it some thought. 👍 
_To set expectations, my current focus is on the v9 migration, (progress updates here: #105)._

**Comment by leekelleher** [comment]
Closing this off, as the idea became the [Data Picker editor](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/data-picker.md).
